os/fs/smartfs/smartfs_smart.c

```diff
@@ -1696,13 +1716,29 @@ static int smartfs_mkdir(struct inode *mountpt, const char *relpath, mode_t mode
 
 		/* Create the directory */
 
-		ret = smartfs_createentry(fs, parentdirsector, filename,
-								  SMARTFS_DIRENT_TYPE_DIR, mode, &entry, 0xFFFF, NULL);
-		if (ret != OK) {
-			goto errout_with_semaphore;
+#ifdef CONFIG_SMARTFS_USE_SECTOR_BUFFER
+		ret = smartfs_find_firstinvalidentry(fs, parentdirsector, &invalidentry);
+		if (ret == OK) {
+			invalidentry.firstsector = SMARTFS_ERASEDSTATE_16BIT;
+			ret = smartfs_replaceentry(fs, invalidentry, filename, SMARTFS_DIRENT_TYPE_DIR, mode, &entry, NULL);
+			if (ret != OK) {
+				goto errout_with_semaphore;
+			}
+		} else {
+			if (ret != -ENOENT) {
```

Comment:
line 292 also can be applied this logic.

Reply:
Done